scalalib/src/mill/javalib/spotless/JavaSpotlessModule.scala

```diff
@@ -0,0 +1,35 @@
+package mill.javalib.spotless
+
+import mill.{Agg, PathRef, T, Task}
+import mill.scalalib.DepSyntax
+
+trait JavaSpotlessModule extends SpotlessModule {
+
+  /**
+   * Classpath for running
+   * - Google Java Format
+   * - Palantir Java Format
+   */
+  def jvmLangLibClasspath: T[Seq[PathRef]] = Task {
+    defaultResolver().resolveDeps(
+      Agg(
+        ivy"com.google.googlejavaformat:google-java-format:${googleJavaFormatVersion()}",
+        ivy"com.palantir.javaformat:palantir-java-format:${palantirJavaFormatVersion()}"
+      )
+    )
+  }
+
+  /**
+   * Google Java Format version. Defaults to `1.25.2`.
+   */
+  def googleJavaFormatVersion: T[String] = Task {
+    "1.25.2"
+  }
+
+  /**
+   * Google Java Format version. Defaults to `2.50.0`.
+   */
+  def palantirJavaFormatVersion: T[String] = Task {
+    "2.50.0"
```

Comment:
Same as for `googleJavaFormatVersion`, versions should not be hardcoded. Have a look at PR https://github.com/com-lihaoyi/mill/pull/4552, which replaces another instance of a hardcoded version.